### ISSUE TITLE
📎 Improve error reporting for plugin syntax errors

### ISSUE THREAD

**Issue by arendjr** [biomejs/biome#5610]
### Description

In Biome 2.0.0-beta.1, if you create a plugin with incorrect syntax, you get an error like this:

```
$ yarn biome check
yarn run v1.22.22
$ /home/arendjr/Projects/ts-playground/node_modules/.bin/biome check
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: /home/runner/work/biome/biome/crates/biome_parser/src/diagnostic.rs:350:14
Thread Name: main
Message: Expected token to be a punctuation or keyword.

error Command failed with exit code 101.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Can be reproduced with the following plugin:

```gritql
jsx_self_closing_element { 
  name: identifier {
    text: `img`
  }
  register_diagnostic(
    span = $name,
    message = "use Image component instead of img tag"
  )
}
```

We need better error reporting for this.
